packages/sycamore-macro/src/lib.rs

```diff
@@ -15,6 +15,27 @@ pub fn template(component: TokenStream) -> TokenStream {
     template::template_impl(component).into()
 }
 
+/// ```
+/// use sycamore::{generic_node::EventHandler, prelude::*};
+///
+/// #[component(MyComponent<G>)]
+/// pub fn my_component(event_listeners: Vec<(String, Box<EventHandler>)>) -> Template<G> {
+///     let cool_button: G = node! { button { "The coolest 😎" } };
+///
+///     for listener in event_listeners {
+///         cool_button.event(listener.0.as_ref(), listener.1);
+///     }
+///
+///     Template::new_node(cool_button)
+/// }
+/// ```
+#[proc_macro]
+pub fn node(input: TokenStream) -> TokenStream {
+    let node = parse_macro_input!(input as template::Element);
+
+    template::node_impl(node).into()
+}
+
```

Comment:
How about we do something else such as setting a property because this is technically bad practice: the listeners will never be dropped which is a memory leak.
```suggestion
/// A macro for creating nodes.
/// ```
/// pub fn my_component() -> Template<G> {
///     let cool_button: G = node! { button { "The coolest 😎" } };
///     cool_button.set_property("my_property", &123.into());
///     Template::new_node(cool_button)
/// }
/// ```
#[proc_macro]
pub fn node(input: TokenStream) -> TokenStream {
    let node = parse_macro_input!(input as template::Element);

    template::node_impl(node).into()
}
```